docs/concepts/how-clerk-works.mdx

```diff
@@ -0,0 +1,111 @@
+
+The `__session` cookie is set on the host application that is secured by Clerk (for example `https://dashboard.<example.com>`). It’s a JS cookie, it’s secure, and it contains a short-lived session JWT that lasts 60 seconds. This JWT contains the current session, user, and organization identifiers and must be sent to the application API to authenticate API requests. In SSR, the `__session` cookie travels automatically to the server. ClerkJS ensures that the `__session` cookie is automatically refreshed in Client Side Rendering (CSR) and Server Side Rendering (SSR), ensuring a fresh session is always available.
+
+However, if a `__session` cookie is expired on a request to an application’s backend, the SDK doesn’t know if the session has ended, or if a new short-lived JWT needs to be issued. When an SDK gets into this state, it triggers Clerk's client handshake.
+
+### The client handshake
+
+Clerk uses a client handshake mechanism to resolve a request’s authentication state from unknown (`handshake`) to `signed-in` or `signed-out`. 
+
+With this handshake, Clerk can resolve authentication state on the backend and ensure the request is properly handled as signed in or out, instead of being in a potentially unknown state. The handshake flow relies on redirects to exchange session information between FAPI and the application, ensuring the resolution of unknown authentication states minimizes performance impact and behaves consistently across different framework and language implementations.
+
+Learn more about [how Clerk's client handshake works.](/docs/concepts/client-handshake).
+
+### Client piggybacking
+
+Every Frontend API request returns a response payload that contains two top level keys: 
+
+```json
+{
+   "client": ...,
+   "response": ...
+}
+```
+
+<Callout type="info">
+  The `/v1/client/sessions/:id/tokens` endpoint refreshes the Clerk session tokens and is an exception to this format.
+</Callout>
+
+
+The `response` key contains the resource or resources of each endpoint. The `client` key contains the client piggybacking payload. The client object identifies the current device and contains all sessions, users, and organization data for the current device (see [Main Objects](#main-objects) section). 
+
+This is effectively the necessary frontend state for ClerkJS and powers the JS SDKs, the [React hooks](/docs/references/react/use-user), and [Clerk components](/docs/components/overview). It is a best-effort way for the backend to update the frontend state on every Frontend API request. 
+
+It's not recommended to use Client piggybacking because of the following issues:
+
+1. It’s a very large chunk of a tree-like state and doesn’t change often.
+1. It’s hard to compute on the backend as it requires a lot of data from the database. The computation takes place on every request, adding a significant overhead.
+1. It’s best effort.
+1. It’s not a two-way communication channel.
+
+### Clerk Frontend API for non-standard web browsers
```

Comment:
Please move this under the Clerk Frontend API section.

Reply:
Also add a link to https://clerk.com/docs/reference/frontend-api similarly to BAPI.